### PR TITLE
net.minecraft.client.gui.screens.*

### PR DESCRIPTION
**Pull request by MelanX**
Only the main packages because there were so many. Sub-packages in seperate PR.
Could be conflicting with #76 because of the one matrixStack -> poseStack. I don't know exactly why this was changed.

### CODE REVIEW

File: data/net/minecraft/client/gui/screens/AccessibilityOptionsScreen.mapping [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CLASS net/minecraft/client/gui/screens/AccessibilityOptionsScreen
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/client/Options;)V
+		ARG 1 lastScreen
+		ARG 2 options
+	METHOD lambda$createFooter$1 (Lnet/minecraft/client/gui/components/Button;)V
+		ARG 1 button
+	METHOD lambda$createFooter$2 (Lnet/minecraft/client/gui/components/Button;)V
+		ARG 1 button
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/AccessibilityOptionsScreen.mapping]: Might be nice if these were named by what their button was so their function is instantly recognisable, i.e. `openGuideButton`, `doneButton`, etc (echo for other button lambdas if you do decide to do this).

File: data/net/minecraft/client/gui/screens/ConfirmLinkScreen.mapping [hunk 2]
```diff
@@ -1,4 +1,7 @@
 CLASS net/minecraft/client/gui/screens/ConfirmLinkScreen
+	METHOD <init> (Lit/unimi/dsi/fastutil/booleans/BooleanConsumer;Ljava/lang/String;Z)V
+		ARG 1 callback
+		ARG 2 url
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/ConfirmLinkScreen.mapping:4]: Could add the remaining boolean param here; my suggestion is `trusted`.

File: data/net/minecraft/client/gui/screens/CreateBuffetWorldScreen.mapping [hunk 3]
```diff
@@ -1,4 +1,13 @@
 CLASS net/minecraft/client/gui/screens/CreateBuffetWorldScreen
+	METHOD <init> (Lnet/minecraft/client/gui/screens/Screen;Lnet/minecraft/core/RegistryAccess;Ljava/util/function/Consumer;Lnet/minecraft/world/level/biome/Biome;)V
+		ARG 1 parent
+		ARG 2 registry
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/CreateBuffetWorldScreen.mapping]: This isn't a registry in itself, it provides access to registries. 
```suggestion
		ARG 2 registryAccess
```

File: data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping [hunk 4]
```diff
@@ -1,10 +1,30 @@
...
 		COMMENT The parent GUI
+	METHOD <init> (Lnet/minecraft/client/gui/screens/CreateFlatWorldScreen;)V
+		ARG 1 parent
+	METHOD fromString (Lnet/minecraft/core/Registry;Ljava/lang/String;Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;)Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;
+		ARG 0 registry
+		ARG 1 settings
+		ARG 2 levelGeneratorSettings
+	METHOD getLayerInfoFromString (Ljava/lang/String;I)Lnet/minecraft/world/level/levelgen/flat/FlatLayerInfo;
+		ARG 0 setting
+		ARG 1 currentLayers
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping]: This appears to be the current height, as opposed to the number of layers.
```suggestion
		ARG 1 currentHeight
```

File: data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping [hunk 5]
```diff
@@ -1,10 +1,30 @@
...
 	FIELD parent Lnet/minecraft/client/gui/screens/CreateFlatWorldScreen;
 		COMMENT The parent GUI
+	METHOD <init> (Lnet/minecraft/client/gui/screens/CreateFlatWorldScreen;)V
+		ARG 1 parent
+	METHOD fromString (Lnet/minecraft/core/Registry;Ljava/lang/String;Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;)Lnet/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings;
+		ARG 0 registry
+		ARG 1 settings
+		ARG 2 levelGeneratorSettings
+	METHOD getLayerInfoFromString (Ljava/lang/String;I)Lnet/minecraft/world/level/levelgen/flat/FlatLayerInfo;
+		ARG 0 setting
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping]: `string` or `layerInfo` would be better.

File: data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping [hunk 6]
```diff
@@ -1,10 +1,30 @@
...
+		ARG 1 currentLayers
+	METHOD getLayersInfoFromString (Ljava/lang/String;)Ljava/util/List;
+		ARG 0 settings
```
Comment (Harleyoc1) [inline data/net/minecraft/client/gui/screens/PresetFlatWorldScreen.mapping]: Again, `string` or `layersInfo`.

**Review by Harleyoc1 — CHANGES_REQUESTED** [review]

**Review by alcatrazEscapee — APPROVED** [review]

**Review by Harleyoc1 — APPROVED** [review]

**Pull request merged**